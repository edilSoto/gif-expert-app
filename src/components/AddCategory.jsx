import { useState } from "react"


const AddCategory = ({onNewCategory}) => {
    
    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;

        onNewCategory(inputValue);
        setInputValue('');
    } 

    return (
        <form onSubmit={onSubmit}>
            <input 
            type='text'
            placeholder="Search your topic"
            value={inputValue}
            onChange={onInputChange}
            />
        </form>
    )
}

export default AddCategory