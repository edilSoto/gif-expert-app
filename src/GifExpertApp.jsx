import { useState } from "react";
import { AddCategory, GifGrid } from './components/';

const GifExpertApp = () => {
    const [categories, setCategories] = useState(['Dragon ball']);

    const onAddCategory = (value) => {

        if (categories.includes(value)) return;
        setCategories([value, ...categories])
    }

    return (
        <>
            <h1>Gif Expert App</h1>
            <AddCategory onNewCategory={(value) => onAddCategory(value)} />
                
                {
                    categories.map((category) => (
                        <GifGrid
                            key={category}
                            category={category}
                        />
                    ))
                }
        </>
    )
}

export default GifExpertApp;

// API key: FrY4lxSQNx8eNGBRnqpwIHd4dI7TMYZr