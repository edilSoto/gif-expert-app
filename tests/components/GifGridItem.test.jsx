import { render, screen } from "@testing-library/react"
import { GifGridItem } from "../../src/components"

describe('Tests in the GifGridItem component', () => {

    const title = 'Dragon Ball';
    const url = 'https://media1.giphy.com/media/977YesTjNfQC7vQiph/giphy.gif?cid=0e87d200z5alyfsv7135ihz623f3wli93uiyn1umo1han6pm&rid=giphy.gif&ct=g'

    test('Must match snapshot', () => {

        const { container } = render(<GifGridItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    });

    test('Must show the img element with the url and alt supplied', () => {

        render(<GifGridItem title={title} url={url} />);
        const { src, alt } = screen.getByRole('img');
        expect(src).toMatch(url);
        expect(alt).toMatch(title);
    })

    test('Must show the title of the component', () => {

        render(<GifGridItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });
});