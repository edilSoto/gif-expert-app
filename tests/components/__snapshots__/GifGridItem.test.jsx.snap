// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in the GifGridItem component Must match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="https://media1.giphy.com/media/977YesTjNfQC7vQiph/giphy.gif?cid=0e87d200z5alyfsv7135ihz623f3wli93uiyn1umo1han6pm&rid=giphy.gif&ct=g"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
